.github/workflows/run-archiver.yml

```diff
@@ -50,38 +47,43 @@ jobs:
           ZENODO_SANDBOX_TOKEN_UPLOAD: ${{ secrets.ZENODO_SANDBOX_TOKEN_UPLOAD }}
           ZENODO_SANDBOX_TOKEN_PUBLISH: ${{ secrets.ZENODO_SANDBOX_TOKEN_PUBLISH }}
           EPACEMS_API_KEY: ${{ secrets.EPACEMS_API_KEY }}
-          # They're in the repository secrets, but we need more QA/QC checks
-          # ZENODO_TOKEN_UPLOAD: ${{ secrets.ZENODO_TOKEN_UPLOAD }}
-          # ZENODO_TOKEN_PUBLISH: ${{ secrets.ZENODO_TOKEN_PUBLISH }}
+          ZENODO_TOKEN_UPLOAD: ${{ secrets.ZENODO_TOKEN_UPLOAD }}
+          ZENODO_TOKEN_PUBLISH: ${{ secrets.ZENODO_TOKEN_PUBLISH }}
         run: |
           pudl_archiver --sandbox --datasets ${{ matrix.dataset }} --summary-file ${{ matrix.dataset }}_run_summary.json
 
-      - name: Archive code coverage results
+      - name: Upload run summaries
+        id: upload_summaries
         uses: actions/upload-artifact@v3
         with:
-          name: $${{ matrix.dataset }}-run-summary
+          name: run-summaries
           path: ${{ matrix.dataset }}_run_summary.json
 
   archive-notify:
     runs-on: ubuntu-latest
     needs: archive-run
     if: ${{ always() }}
     steps:
+      - uses: actions/checkout@v4
+      - name: Download summaries
+        id: download
+        uses: actions/download-artifact@v3
+        with:
+          name: run-summaries
+      - name: show summaries
+        run: ls -R
+      - name: Munge summaries together
+        id: all_summaries
+        run: |
+          {
+            echo "SLACK_PAYLOAD<<EOF"
+            ./scripts/make_slack_notification_message.py --summary-files *_run_summary.json | tee slack-payload.json
+            echo "EOF"
+          } >> "$GITHUB_OUTPUT"
       - name: Inform the Codemonkeys
-        uses: 8398a7/action-slack@v3
+        uses: 8398a7/action-slack@v3 # uses unofficial slack action because official one doesn't handle payload correctly https://github.com/slackapi/slack-github-action/issues/261
         with:
           status: custom
-          fields: workflow,job,commit,repo,ref,author,took
-          custom_payload: |
-            {
-              username: 'action-slack',
-              icon_emoji: ':octocat:',
-              attachments: [{
-                color: '${{ needs.archive-run.result }}' === 'success' ? 'good' : '${{ needs.archive-run.result }}' === 'failure' ? 'danger' : 'warning',
-                text: `${process.env.AS_REPO}@${process.env.AS_REF}\n ${process.env.AS_WORKFLOW} (${process.env.AS_COMMIT})\n by ${process.env.AS_AUTHOR}\n Status: ${{ needs.archive-run.result }}`,
-              }]
-            }
+          custom_payload: ${{ steps.all_summaries.outputs.SLACK_PAYLOAD }}
         env:
-          GITHUB_TOKEN: ${{ github.token }} # required
           SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }} # required
```

Comment:
This needs to change once we get the #pudl-deployments webhook thing sorted, but it's just sending to codemonkeys now.